File: src/SubtitleEditor.Web/Typescripts/App/src/elements/models/topic/TopicSubtitleData.ts
import { SubtitleData } from "SubtitleEditor";

export default interface TopicSubtitleData {
    name: string;
    filename: string;
    extension: string;
    description?: string;
    subtitle: SubtitleData;
    transcript?: string;
    frameRate?: number;
    wordLimit?: number;
}

File: src/SubtitleEditor.Web.Infrastructure/Models/Topic/TopicSubtitleData.cs
using SubtitleEditor.Core.Models;

namespace SubtitleEditor.Web.Infrastructure.Models.Topic;

public class TopicSubtitleData
{
    public string Name { get; set; } = string.Empty;
    public string Filename { get; set; } = string.Empty;
    public string Extension { get; set; } = string.Empty;
    public string? Description { get; set; }
    public Subtitle Subtitle { get; set; } = null!;
    public string? Transcript { get; set; }
    public double? FrameRate { get; set; }
    public int? WordLimit { get; set; }
}